consensus/polybft/state_sync_manager.go

```diff
@@ -409,38 +423,49 @@ func (s *stateSyncManager) buildProofs(commitmentMsg *CommitmentMessage) error {
 }
 
 // buildCommitment builds a new commitment, signs it and gossips its vote for it
-func (s *stateSyncManager) buildCommitment(epoch, fromIndex uint64) (*Commitment, error) {
-	toIndex := fromIndex + stateSyncCommitmentSize - 1
+func (s *stateSyncManager) buildCommitment() error {
+	s.lock.RLock()
 
-	// if it is not already built in the previous epoch
-	stateSyncEvents, err := s.state.getStateSyncEventsForCommitment(fromIndex, toIndex)
+	epoch := s.epoch
+	fromIndex := s.nextCommittedIndex
+
+	var largestPendingCommitment *Commitment
+	if len(s.pendingCommitments) > 0 {
+		largestPendingCommitment = s.pendingCommitments[len(s.pendingCommitments)-1]
+	}
+
+	s.lock.RUnlock()
+
+	stateSyncEvents, err := s.state.getStateSyncEventsForCommitment(fromIndex, fromIndex+maxCommitmentSize-1, false)
 	if err != nil {
-		if errors.Is(err, errNotEnoughStateSyncs) {
-			s.logger.Debug("[buildCommitment] Not enough state syncs to build a commitment",
-				"epoch", epoch, "from state sync index", fromIndex)
+		return fmt.Errorf("failed to get state sync events for commitment. Error: %w", err)
+	}
 
-			// this is a valid case, there is not enough state syncs
-			return nil, nil
-		}
+	if len(stateSyncEvents) < minCommitmentSize {
+		// there is not enough state sync events to build at least the minimum commitment
+		return nil
+	}
 
-		return nil, err
+	if largestPendingCommitment != nil && largestPendingCommitment.ToIndex >= stateSyncEvents[len(stateSyncEvents)-1].ID {
```

Comment:
If `largestPendingCommitment` is used only for this check can we move get read of variable on stack and do something like:
```
	if len(s.pendingCommitments) > 0 &&
            s.pendingCommitments[len(s.pendingCommitments)-1].ToIndex >= stateSyncEvents[len(stateSyncEvents)-1].ID { 
            ...
	}
```

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1083/commits/7473df4bf9f724b43e552512a25260e8ead65fa2